tests/mapobject/test_mapobject.cpp

```diff
@@ -0,0 +1,44 @@
+#include "mapobject.h"
+#include "objecttemplate.h"
+
+#include <QtTest/QtTest>
+
+using namespace Tiled;
+
+class test_MapObject : public QObject
+{
+    Q_OBJECT
+
+private slots:
+    void test_syncWithTemplate();
+};
+
+
+void test_MapObject::test_syncWithTemplate()
+{
+    MapObject obj;
+    Properties props;
+    props["hp"] = 50;
+    obj.addComponent("component1", props);
+    ObjectTemplate templateObj;
+    templateObj.setObject(&obj);
+
+    MapObject mapObject;
+    mapObject.setObjectTemplate(&templateObj);
+
+    mapObject.syncWithTemplate();
+    QCOMPARE(mapObject.componentProperties("component1")["hp"], 50);
```

Comment:
This appears to cause:
```
undefined reference to `bool QTest::qCompare<QVariant, int>(QVariant const&, int const&, char const*, char const*, char const*, int)'
```
when compiling against Qt 5.6 with MinGW 4.9 (the WinXP build). Maybe replace `50` with `QVariant(50)`.